lib/versioning/rez/index.ts

```diff
@@ -0,0 +1,324 @@
+import { api as npm } from '../npm';
+import { api as pep440 } from '../pep440';
+import type { NewValueConfig, VersioningApi } from '../types';
+
+export const id = 'rez';
+export const displayName = 'rez';
+export const urls = ['https://github.com/nerdvegas/rez'];
+export const supportsRanges = true;
+export const supportedRangeStrategies = ['bump', 'extend', 'pin', 'replace'];
+// Regular Expressions have been copied from, some more work were necessary to make it work:
+// https://github.com/nerdvegas/rez/blob/master/src/rez/vendor/version/version.py
```

Comment:
License for the copied work is here:

https://github.com/nerdvegas/rez/blob/master/COPYING.LESSER

I don't know if copy/pasting and adjusting like this is allowed under our and their license?

Reply:
🤔 I'm not sure if a regex expression is protectable, but we should ask them to be sure.

nerdvegas/rez#1110

Reply:
Very good question, did not think of that. @viceice, thanks for creating the issue officially right away.

Hope it's not a problem, would be unnecessarily hard to come up with different logic for this.

Reply:
We got an answer. Please add a clear comment that some regexes are from rez and have a GPL license.

@rarkins Is it ok to integrate that code?

Reply:
Unfortunately it's not OK, unless we had a way to remove rez versioning from commercial editions downstream. WhiteSource Renovate On-Premises for example is EULA-licensed and although it's free, we can't bundle others GPL code within it and so could no longer offer it.

What we would need to do is have someone new to this - so unfortunately not @skral or any of us in this thread now - implement our own parsing based off a spec and completely independently of rez's own source code and without having seen the regexes. If our rez versioning is in no way derived from their GPL'd code, then there's less chance of a problem.

The alternative would be that whomever contributed to the regexes part of their code grants us a permissive license for their use but that's perhaps unlikely.

Final alternative: we finally implement a "plugin" architecture so that Renovate users can more easily add in their own custom versioning/managers/datasources, in which case rez could exist as a GPL'd Renovate plugin, easily combinable with Renovate in a custom build.

Reply:
> The alternative would be that whomever contributed to the regexes part of their code grants us a permissive license for their use but that's perhaps unlikely.

The original author (i.e. nerdvegas - via git blame of the code section) of the regular expressions we're talking about has said "If all you need to do is grab some regexes, I would just do that, and include a comment linking to the rez project.". That's not a license but isn't this the ok that is needed to copy the expressions?

> implement our own parsing based off a spec and completely independently of rez's own source code and without having seen the regexes

That is of course doable, I can find someone to help with that but I might as well approach the problem completely different and scrap the expressions.

Let me know what you think is the best way forward.

Reply:
I think the response you got was in the context of "Renovate is AGPL, can we use your GPL regexes?" and not in the context I described afterwards, which is that we have a requirement to have rights to use code in a non-GPL way. If we were to reuse the regexes in the way we need, the best way I can think of to do that is if the original author submits a (non-functional) PR here simply containing the required regex snippets and completing our CLA. This would provide us the grant we need to reuse the regex while not having any impact on the remainder of rez.

And to clarify:
- I don't think the original author is under any obligation at all to do this, it's purely their choice
- The reason they'd want to do this is because the benefits of improving the rez ecosystem outweigh any negatives of granting Renovate/WhiteSource rights to use the snippets how they like

Reply:
Asked the original author for help with creating the PR [here](https://github.com/nerdvegas/rez/issues/1110#issuecomment-891271552). We'll see what he says.

Reply:
We should be good here now. Thanks to @nerdvegas.